### PR TITLE
system_var, planner: add variable for controlling the algorithm of jo…

### PR DESCRIPTION
**Pull request by winoros**
…in reorder

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add the description for the variable to control the join reorder algo.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1732,6 +1732,15 @@ mysql> desc select count(distinct a) from test.t;
     select t.* from t, t1 where t.a=t1.aa;
     ```
 
+### `tidb_opt_join_reorder_threshold`
```
Comment (shichun-0415) [inline system-variables.md:1735]:
```suggestion
### `tidb_opt_join_reorder_threshold` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```

File: system-variables.md [hunk 2]
```diff
@@ -1732,6 +1732,15 @@ mysql> desc select count(distinct a) from test.t;
...
+
+- 作用域: SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0`
+- 范围：`[0, 2147483647]`
+- 这个变量用来控制 TIDB Join Reorder 算法的选择
+- 目前对于 OLTP 的查询推荐保持默认值，对于 OLAP 的查询推荐将其设置为 10~15 来获得 AP 场景下更好的连接顺序。
```
Comment (qiancai) [inline system-variables.md:1742]: 默认值代表不选择 TiDB Join Reorder 算法吗？

Comment (winoros) [reply system-variables.md:1742]: 是二选一的关系

File: system-variables.md [hunk 3]
```diff
@@ -1732,6 +1732,15 @@ mysql> desc select count(distinct a) from test.t;
     select t.* from t, t1 where t.a=t1.aa;
     ```
 
+### `tidb_opt_join_reorder_threshold` <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (qiancai) [inline system-variables.md:1735]:
```suggestion
### `tidb_opt_join_reorder_threshold`
```

File: system-variables.md [hunk 4]
```diff
@@ -1732,6 +1732,15 @@ mysql> desc select count(distinct a) from test.t;
     select t.* from t, t1 where t.a=t1.aa;
     ```
 
+### `tidb_opt_join_reorder_threshold`
+
+- 作用域: SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0`
+- 范围：`[0, 2147483647]`
+- 这个变量用来控制 TiDB Join Reorder 算法的选择。
```
Comment (TomShawn) [inline system-variables.md:1741]: 是小于阈值会使用 Join Reorder 算法吗？还是大于阈值会使用呢？

Comment (winoros) [reply system-variables.md:1741]: 是两种不同的 join reorder 算法，大于这个值的时候选贪心，小于的选另一种。另一种的文档我在这个 sprint 补上。

Comment (TomShawn) [reply system-variables.md:1741]:
```suggestion
- 这个变量用来控制 TiDB Join Reorder 算法的选择。当参与 Join Reorder 的节点个数大于该阈值时，TiDB 选择贪心算法，小于该阈值时 TiDB 选择动态规划 (dynamic programming) 算法。
```

File: system-variables.md (hunk 2 above)
Comment (TomShawn) [reply system-variables.md:1742]:
```suggestion
- 目前对于 OLTP 的查询，推荐保持默认值。对于 OLAP 的查询，推荐将变量值设为 10~15 来获得 AP 场景下更好的连接顺序。
```

**Review by fixdb — APPROVED** [review]
+1

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**